graspologic/plot/plot.py

```diff
@@ -671,6 +676,226 @@ def pairplot(
+
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        dimensions = X.shape[1]
+        # we only want 1 scatter plot for 2 features
+        if X.shape[1] == 2:
+            dimensions = 1
+        pairplot, axes = plt.subplots(
+            dimensions, dimensions, figsize=figsize, squeeze=False
+        )
+        # this will allow for uniform iteration whether axes was 2d or 1d
+        axes = axes.flatten()
+        for i in range(dimensions):
+            for j in range(dimensions):
+                if i == j and X.shape[1] > 2:
+                    # take care of the distplot on diagonal
+                    for t, lab in zip([i for i in range(X.shape[1])], label_palette):
+                        sns.distplot(
+                            X[Y_ == t, i],
+                            kde=kde,
+                            ax=axes[dimensions * i + j],
+                            color=label_palette[lab],
+                        )
+                    # this removes the tick marks from the histplot
+                    axes[dimensions * i + j].set_xticks([])
+                    axes[dimensions * i + j].set_yticks([])
+                else:
+                    # take care off off-diagonal scatterplots
+                    ax1, ax2 = j, i
+                    # with only a scatter plot we must make sure we plot
+                    # the first and second feature of X
+                    if X.shape[1] == 2:
+                        ax1, ax2 = 0, 1
+                    _plot_ellipse_and_data(
+                        data,
+                        X,
+                        ax1,
+                        ax2,
+                        means,
+                        covariances,
+                        axes[dimensions * i + j],
+                        label_palette,
+                        cluster_palette,
+                        alpha=alpha,
+                    )
+        # formatting
+        if title:
+            pairplot.suptitle(title)
+
+        for i in range(dimensions):
+            for j in range(dimensions):
+                if X.shape[1] == 2:
+                    axes[dimensions * i + j].set_ylabel("Dimension " + str(0))
+                    axes[dimensions * i + j].set_xlabel("Dimension " + str(1))
+                else:
+                    axes[dimensions * i + j].set_ylabel("Dimension " + str(i))
+                    axes[dimensions * i + j].set_xlabel("Dimension " + str(j))
+
+        # we get the handels (like the colored dot in legend)
+        # we get the associated label
+        # in order to show only the unique pairs we do the following
+        handles, labels = [], []
+        for ax in axes.flat:
+            ax.label_outer()
+            ax.spines["right"].set_visible(False)
+            ax.spines["top"].set_visible(False)
+            handles_, labels_ = ax.get_legend_handles_labels()
+            handles += handles_
+            labels += labels_
```

Comment:
why is it necessary to get handles/labels from each axis? If you got them from one axis, wouldn't they be correct for all? If not, seems like a bug. If so, seems like that would solve this issue?

Reply:
https://github.com/microsoft/graspologic/pull/541/commits/72d227c50777c95a9785ffe59d1cb292bae3b9b5